kernel/api/src/main/java/org/sakaiproject/site/api/SiteService.java

```diff
@@ -1203,4 +1203,18 @@ public boolean isAsc()
 	 */
 	public String getParentSite(String siteId);
 
+	/**
+	 * Given a site and a list of provider IDs for the site, return the appropriate
+	 * site or section title.
+	 * 
+	 * SAK-29138 - Takes into account 'portal.use.sectionTitle' sakai.property; 
+	 * if set to true, this method will return the title of the section the current 
+	 * user is enrolled in for the site (if it can be found). Otherwise, it will 
+	 * return the site title (default behaviour).
+	 * 
+	 * @param site the site in question
+	 * @param providerIDs the list of provider IDs for the site
+	 * @return the site or section title
+	 */
+	public String getSiteOrSectionTitle( Site site, List<String> providerIDs );
```

Comment:
Why would you ever need pass in providerIDs that didn't belong to the site passed as the first argument?